js/util/browser/browser.js

```diff
@@ -98,7 +100,14 @@ exports.supported = function(options) {
         },
 
         function() {
-            return new Canvas().supportsWebGLContext((options && options.failIfMajorPerformanceCaveat) || false);
+            var opt = (options && options.failIfMajorPerformanceCaveat) || false,
+                fimpc = 'fimpc_' + String(opt);
+            if (exports.supportsWebGL[fimpc] === undefined) {
+                var canvas = new Canvas();
+                exports.supportsWebGL[fimpc] = canvas.supportsWebGLContext(opt);
+                canvas.canvas = undefined;
```

Comment:
This will have no effect; `canvas` is going out of scope anyway and thus `canvas.canvas` and `canvas` are already equally eligible for GC. If `canvas` had an API for forced finalization we could use that, but I don't see one.